protos/local/core_interface.proto

```diff
@@ -16,12 +16,28 @@ import "google/protobuf/timestamp.proto";
 import "google/protobuf/duration.proto";
 import "google/protobuf/empty.proto";
 
-// A request as given to [crate::Core::send_activity_heartbeat]
+// A request as given to [crate::Core::record_activity_heartbeat]
 message ActivityHeartbeat {
-    string activity_id = 1;
+    ActivityId activity_id = 1;
     repeated common.Payload details = 2;
 }
 
+message ActivityId {
+    oneof id {
+        bytes task_token = 1;
+        ManualCompletionActivityTaskToken manual_completion_task_token = 2;
+    }
+}
+
+message ManualCompletionActivityTaskToken {
+    common.WorkflowExecution workflow_execution = 1;
+    string activity_id = 2;
+}
+
+message RecordActivityHeartbeatResponse {
+    bool cancel_requested = 1;
```

Comment:
I thought that we're notifying lang of activity cancellation in `ActivityTask`, that's at least what we've talked about.

Reply:
The existing SDKs let you catch errors thrown by heartbeating which can include being cancelled. I think we need that, so then you'd have to block on sending a heartbeat until you get an `ActivityTask` that says it worked or not, which feels like a kinda roundabout way of doing it

Reply:
I think we should discuss this, let me schedule a meeting to clear everything out.

Reply:
Update: after offline conversation we've decided to send cancellation as an activity task to lang. This would allow us innovate more in the area of activity cancellations and to dispatch cancellations without heartbeats in the future.